packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp

```diff
@@ -2407,6 +2228,86 @@ namespace Tpetra {
   }
 
 
+  template <class LocalOrdinal, class GlobalOrdinal, class Node>
+  size_t
+  CrsGraph<LocalOrdinal, GlobalOrdinal, Node>::
+  findLocalIndices(const RowInfo& rowInfo,
+                   const Teuchos::ArrayView<const LocalOrdinal>& indices,
+                   std::function<void(const size_t, const size_t, const size_t)> fun) const
+  {
+    const char tfecfFuncName[] = "findLocalIndices: ";
+    TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC(
+        this->getProfileType() != StaticProfile,
+        std::runtime_error,
+        "findLocalIndices requires the graph have StaticProfile");
+
+    using LO = LocalOrdinal;
+    using Kokkos::View;
+    using Kokkos::MemoryUnmanaged;
+    using inp_view_type = View<const LO*, execution_space, MemoryUnmanaged>;
+    inp_view_type inputInds(indices.getRawPtr(), indices.size());
+
+    size_t numFound = 0;
+    LO lclRow = rowInfo.localRow;
+    if (this->isLocallyIndexed())
+    {
+      numFound = Details::findCrsIndices(lclRow, k_rowPtrs_,
+        this->k_lclInds1D_, inputInds, fun);
+    }
+    else if (this->isGloballyIndexed())
+    {
+      if (this->colMap_.is_null())
+        return Teuchos::OrdinalTraits<size_t>::invalid();
+      const auto& colMap = *(this->colMap_);
+      auto map = [&](LO const lclInd){return colMap.getGlobalElement(lclInd);};
+      numFound = Details::findCrsIndices(lclRow, k_rowPtrs_,
+        this->k_gblInds1D_, inputInds, map, fun);
+    }
+    return numFound;
+  }
+
+
+  template <class LocalOrdinal, class GlobalOrdinal, class Node>
+  size_t
+  CrsGraph<LocalOrdinal, GlobalOrdinal, Node>::
+  findGlobalIndices(const RowInfo& rowInfo,
+                    const Teuchos::ArrayView<const GlobalOrdinal>& indices,
+                    std::function<void(const size_t, const size_t, const size_t)> fun) const
+  {
+    const char tfecfFuncName[] = "findGlobalIndices: ";
+    TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC(
+        this->getProfileType() != StaticProfile,
+        std::runtime_error,
+        "findLocalIndices requires the graph have StaticProfile");
+
+    using GO = GlobalOrdinal;
+    using Kokkos::View;
+    using Kokkos::MemoryUnmanaged;
+    auto invalidCount = Teuchos::OrdinalTraits<size_t>::invalid();
+
+    using inp_view_type = View<const GO*, execution_space, MemoryUnmanaged>;
+    inp_view_type inputInds(indices.getRawPtr(), indices.size());
+
+    size_t numFound = 0;
+    LocalOrdinal lclRow = rowInfo.localRow;
+    if (this->isLocallyIndexed())
+    {
+      if (this->colMap_.is_null())
+        return invalidCount;
+      const auto& colMap = *(this->colMap_);
+      auto map = [&](GO const gblInd){return colMap.getLocalElement(gblInd);};
+      numFound = Details::findCrsIndices(lclRow, k_rowPtrs_,
```

Comment:
Hmmm, `Details::findCrsIndices` is templated on the `map` type.  Are you seeing a different version of the code?

Reply:
`Details::insertCrsIndices` is *not* templated on the `map` type since it leads to an ambiguous function overload set since the callback function is optional.